test/modules/DirectMethodSender/DirectMethodSenderBase.cs

```diff
@@ -52,6 +52,11 @@ abstract class DirectMethodSenderBase : IDisposable
                 logger.LogInformation($"Invoke DirectMethod with count {this.directMethodCount}: finished.");
                 return new Tuple<HttpStatusCode, ulong>((HttpStatusCode)resultStatus, this.directMethodCount);
             }
+            catch (UnauthorizedException e)
+            {
+                logger.LogInformation(e, $"Unauthorized exception caught with count {this.directMethodCount}");
+                return new Tuple<HttpStatusCode, ulong>(HttpStatusCode.Unauthorized, this.directMethodCount);
+            }
```

Comment:
Reminder to adjust logic here:
https://github.com/Azure/iotedge/blob/master/test/modules/TestResultCoordinator/Reports/DirectMethod/Connectivity/DirectMethodConnectivityReportGenerator.cs#L264-L289